File: server/app/Models/Ukm.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Ukm extends Model
{
    use HasFactory;

    protected $fillable = [
        'nama_ukm', 'deskripsi', 'logo_url', 'kategori', 'is_pendaftaran_buka', 'user_id',
    ];

    protected $casts = [
        'is_pendaftaran_buka' => 'boolean', 
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    public function admin()
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function kegiatans()
    {
        return $this->hasMany(Kegiatan::class);
    }

    public function anggota()
    {
        return $this->belongsToMany(User::class, 'keanggotaans')->withPivot('status')->withTimestamps();
    }

    
}

File: client/lib/models/ukm.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:client/models/kegiatan.dart';
import 'package:client/models/user.dart';

part 'ukm.g.dart';

@JsonSerializable(explicitToJson: true)
class Ukm {
  final int id;
  @JsonKey(name: 'nama_ukm')
  final String namaUkm;
  final String? deskripsi;
  @JsonKey(name: 'logo_url')
  final String? logoUrl;
  final String? kategori;
  @JsonKey(name: 'is_pendaftaran_buka', defaultValue: false)
  final bool isPendaftaranBuka;
  final User? admin;
  @JsonKey(defaultValue: [])
  final List<Kegiatan>? kegiatans;
  @JsonKey(defaultValue: [])
  final List<User>? anggota;

  Ukm({
    required this.id,
    required this.namaUkm,
    this.deskripsi,
    this.logoUrl,
    this.kategori,
    required this.isPendaftaranBuka,
    this.admin,
    this.kegiatans,
    this.anggota,
  });

  Ukm copyWith({
    int? id,
    String? namaUkm,
    String? deskripsi,
    String? logoUrl,
    String? kategori,
    bool? isPendaftaranBuka,
    User? admin,
    List<Kegiatan>? kegiatans,
    List<User>? anggota,
  }) {
    return Ukm(
      id: id ?? this.id,
      namaUkm: namaUkm ?? this.namaUkm,
      deskripsi: deskripsi ?? this.deskripsi,
      logoUrl: logoUrl ?? this.logoUrl,
      kategori: kategori ?? this.kategori,
      isPendaftaranBuka: isPendaftaranBuka ?? this.isPendaftaranBuka,
      admin: admin ?? this.admin,
      kegiatans: kegiatans ?? this.kegiatans,
      anggota: anggota ?? this.anggota,
    );
  }

  factory Ukm.fromJson(Map<String, dynamic> json) => _$UkmFromJson(json);
  Map<String, dynamic> toJson() => _$UkmToJson(this);
}